src/platform/plugins/shared/saved_search/common/types.ts

```diff
@@ -98,3 +98,43 @@ export type SavedSearch = Partial<SavedSearchAttributes> & {
 export type SerializableSavedSearch = Omit<SavedSearch, 'searchSource'> & {
   serializedSearchSource?: SerializedSearchSourceFields;
 };
+
+export interface DiscoverSessionTab {
+  id: string;
+  label: string;
+  sort: SortOrder[];
+  columns: string[];
+  grid: DiscoverGridSettings;
+  hideChart: boolean;
+  isTextBasedQuery: boolean;
+  usesAdHocDataView?: boolean;
+  serializedSearchSource: SerializedSearchSourceFields;
+  viewMode?: VIEW_MODE;
+  hideAggregatedPreview?: boolean;
+  rowHeight?: number;
+  headerRowHeight?: number;
+  timeRestore?: boolean;
+  timeRange?: Pick<TimeRange, 'from' | 'to'>;
+  refreshInterval?: RefreshInterval;
+  rowsPerPage?: number;
+  sampleSize?: number;
+  breakdownField?: string;
+  density?: DataGridDensity;
+  visContext?: VisContextUnmapped;
+}
+
+export interface DiscoverSession {
```

Comment:
The new `DiscoverSession` model is fully serializable (no resolved search source instance) so it can be stored in Redux. It will also mean no more weird `Serializable` conditional type stuff when we start using it in the embeddable.